### ISSUE TITLE
Not show Challenges that are not deployed.

### ISSUE THREAD

**Issue by Waterman1001** [BentoBoxWorld/Challenges#175]
#### Is your feature request related to a problem?

Players currently are still able to see challenges that are not deployed.

#### Describe the solution you'd like us to implement.

It would be nice to have an option, just like the biomes addon, such that people are not able to see those challenges anymore when they are not deployed.

**Comment by wellnesscookie** [comment]
I partially disagree, as we actually use this. We add a few challenges, but we "lock" them on purpose.
We say that the challenges will be unlocked at that DD.MM.YYYY in HH:MM and we turn it on for a limited time. Until we had previously unlocked them, we let players collect resources for that specific challenge. Toggle option would be more awesome, and then the both sides are satisfied. 

Feauture I'd like to be implemented:
- Toggle whether deployed challenges are showed or not

**Comment by wellnesscookie** [comment]
If this would be toggleable, then yes, I'd like this too.
We use locked (not deployed) challenges and display on purpose for some things.

**Comment by BONNe** [comment]
This feature was not intended to use it like that @wellnesscookie 
:@ :@ :@

:)

**Comment by wellnesscookie** [comment]
Buut, it's really cool to have it! If you would let me try to explain what we do:

We implement a challenge and make it undeployed. We write that the challenge will unlock on DD.MM.YYYY in HH:MM and our players prepare for it. We say them that it will last only for 2 hours, so they prepare as many materials as they can to complete it.

It gets our players hyped and interested in always checking the GUI for something new! :)

**Comment by BONNe** [comment]
oOo,
If you write FR, I could implement it so it becomes deployed automatically when time is reached.

**Comment by wellnesscookie** [comment]
Yes, but it would be kinda useless if you make it invisible when undeployed. We want them to see it but not do it as well

**Comment by BONNe** [comment]
When i will implement it, it will be the same as it is in Biomes.
- Visible,
- Invisible,
- Toggle.

**Comment by BONNe** [comment]
Should undeployed challenges count in the uncompleted challenge list that prevents from unlocking next level?

**Comment by wellnesscookie** [comment]
As for me, yes.

**Comment by Waterman1001** [comment]
For me, it doesn't really matter.
I know that in the current situation they do count. For this reason I configured it this way.
But I can change my configuration if needed since I think it makes more sense that if a challenge is not deployed and thus kind of 'unavailable', it also does not count as an uncompleted challenge.
I do understand the opinion of wellnesscookie because of the way he uses the feature

**Comment by BONNe** [comment]
Partially implemented in build #365
Not sure how to implement the toggleable option.

**Comment by tastybento** [comment]
Status check: the setting exists today as `gui-settings.undeployed-view-mode` with three modes — VISIBLE (show everything), HIDDEN (undeployed challenges don't appear in the player GUI at all), and TOGGLEABLE. VISIBLE and HIDDEN both work, so the original ask here is covered by setting HIDDEN.

However, TOGGLEABLE is still marked "Currently not implemented" in config.yml — selecting it behaves like VISIBLE because the player panel only filters on HIDDEN. I'm going to finish that: a button in the player GUI that lets each player show/hide undeployed challenges themselves when the mode is TOGGLEABLE.

In-game test plan: set `undeployed-view-mode: TOGGLEABLE`, create one deployed and one undeployed challenge, verify the toggle button appears and hides/shows the undeployed one per player; verify VISIBLE and HIDDEN still behave as before.

**Comment by tastybento** [comment]
Implemented the TOGGLEABLE mode in #417.

Setting `undeployed-view-mode: TOGGLEABLE` now adds a toggle button (spyglass, in the free-challenges row) to the player GUI. Each player uses it to show or hide undeployed challenges for themselves. It defaults to *shown*, so TOGGLEABLE starts out just like VISIBLE — which fits wellnesscookie's "hype the upcoming challenge" use case — and a player who prefers a tidy list can hide them. VISIBLE and HIDDEN are unchanged; the button only appears in TOGGLEABLE mode.

The filtering (HIDDEN, TOGGLEABLE-shown, TOGGLEABLE-hidden) is covered by JUnit tests. Since the button itself is template-driven, a quick in-game sanity check:

1. `undeployed-view-mode: TOGGLEABLE`, create one deployed + one undeployed challenge.
2. Open `/challenges` — both show; the toggle button reads "shown".
3. Click it — the undeployed challenge disappears; button reads "hidden". Click again — it comes back.
4. Switch to VISIBLE and HIDDEN and confirm they behave as before and the button is gone.

One note for existing servers: the button lives in `panels/main_panel.yml`, so an already-copied template needs regenerating (delete the file) or the `TOGGLE_UNDEPLOYED` button added by hand — the usual story for template changes. Fresh installs get it automatically.
